### DISCUSSION TITLE
Can't install extension on Rider.

### DISCUSSION THREAD

**Discussion by tatoforever** [community/community#6663] [Copilot Conversations]
Can't install extension on Rider. After adding the link to the repository list, it gives an error: "Connection failed".

**Comment by ffMathy** [comment]
Me too. This is a big deal. Should be easy to enable for Rider too, right? It's the same IDE plugin model as IntelliJ.

**Reply by citizenmatt** [reply]
I can't speak for GitHub, but at a technical level, Rider is rather different to the other IntelliJ IDEs, and it's not always possible to run plugins across all IDEs. Traditionally, IntelliJ based IDEs are JVM applications, with all language processing done in-process, using standard APIs that IntelliJ has created, which can be used to build other features on top (e.g. rename support can be shared across languages because they work to common APIs that each language can implement).

Rider is different, because it's language processing comes from an external process - namely ReSharper, running as an out of process .NET based service. This allows Rider to take advantage of 20 years worth of IntelliJ development, and about 16 years worth of ReSharper dev. But this means that the architecture or Rider is in some ways very different to other IntelliJ IDEs, and not all plugins that can run on multiple IntelliJ IDEs can run on Rider.

**Reply by ffMathy** [reply]
Thank you for this explanation. While it makes me super sad, it really helps understand why they didn't just enable it.

**Comment by joel-matthias** [comment]
It appears that there are differences between the various Jetbrains IDEs that prevent the current Jetbrains CoPilot plug in from working. You can manually install the plugin from disk but then you will see an error about incompatible modules and the plugin will not work. I think we just have to wait for support for Rider and other Jetbrains IDEs to be added (CLion is another IDE that doesn't work with the current CoPilot plugin from what I can tell).

**Reply by joeldg** [reply]
Where can you download the plugin package at?

**Reply by joeldg** [reply]
found it at https://plugins.jetbrains.com/plugin/17718-github-copilot/versions
